web/html/src/manager/admin/payg-shared/api/payg-actions-api.ts

```diff
@@ -0,0 +1,94 @@
+import { useState } from "react";
+
+import { Cancelable } from "utils/functions";
+import Network from "utils/network";
+
+type returnUsePaygActionsApi = {
+  onAction: (actionBodyRequest: any, action: NetworkActionKey, id?: String | null) => Promise<any>;
+  cancelAction: () => void;
+  isLoading: boolean;
+};
+
+const networkAction = {
```

Comment:
Instead of duplicating all of this logic from `use-lifecycle-actions-api.ts`, I would recommend lifting it up instead. Since using an api this way looks like a common pattern, I would recommend making it a generic hook, akin to `usePrevious`. Feel free to ping me if you'd like some help with this.

Reply:
Sure, I think that makes sense yeah. I think it makes sense to assign this ticket to me, the only downside is that I won't know how to test this well. So probably I will bug you for a review and input when I will work on it.

Reply:
Sure that also makes sense. If you don't mind to do the refactoring that would be great. I'm happy to help in anything or even pair program and take the opportunity to learn more.
For the test, I have added some cucumber test, which should catch any problem on the UI.
I will create the card after this PR get merged.